tools/debugging/stress_test_transfers.py

```diff
@@ -152,10 +157,25 @@ def subprocess_stopped(result: AsyncResult) -> None:
 
                     # if the subprocess error'ed propagate the error.
                     if result.get() != STATUS_CODE_FOR_SUCCESS:
-                        self.stop.set()
+                        log.error("Raiden died! Bailing out.")
+                        stop.set()
 
                 process.result.rawlink(subprocess_stopped)
 
+            @staticmethod
+            def spawn_under_watch(function: Callable, *args: Any, **kwargs: Any) -> Greenlet:
+                greenlet = gevent.spawn(function, *args, **kwargs)
+
+                # The Event.rawlink is executed inside the Hub thread, which
+                # does validation and *raises on blocking calls*, to go around
+                # this a new greenlet has to be spawned, that in turn will
+                # raise the exceptoin.
+                def spawn_to_kill() -> None:
+                    gevent.spawn(greenlet.throw, gevent.GreenletExit())
```

Comment:
Do we need to kill the greenlets? We send SIGINT to all processes on exit, so we could cleanly wait for all processes to finish.

Reply:
Hum, this is not strictly necessary.

I introduced this to fix a particular problem, what happened is that `run_stress_test` was not checking the `stop` event, so it was trying to send HTTP requests to the offline node. I don't fully understand why, but that lead the script hanging. My theory was read timeout on the sockets. (It happened because I forgot to open the VPN, all the sub-processes were already dead).

Since IO operations are monkey patched, AFAIK there is no way to cancel it while it is in flight without a monitoring thread, which is pretty much what I implemented here.

The other thing that could be done is to not use the nursery to spawn and track greenlets, but just have the top-level function call on a separate greenlet and a wait on that function and the stop event. I thought that using the concept of the nursery that was already available was a good idea.